src/main/java/com/bootcamp/demo/service/ScooterService.java

```diff
@@ -0,0 +1,77 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Scooter;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+@Service
+public class ScooterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterService.class);
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private final Firestore db;
+
+    private ScooterService(Firestore db) {
+        this.db = db;
+    }
+
+    public Scooter findScooterById(String scooterId) {
+        try {
+            DocumentSnapshot scooter = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooterId)
+                    .get()
+                    .get();
+
+            if (!scooter.exists()) {
+                LOGGER.info("FIND SCOOTER BY ID - service function: Scooter does not exists! scooterId = {}", scooterId);
+                return null;
+            }
+
+            return scooter.toObject(Scooter.class);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("FIND SCOOTER BY ID - service function: scooterId = {}.", scooterId, e);
+            throw new ServiceException(e);
+        }
+    }
+
+    public Set<Scooter> findAllScooters() {
+        try {
+            return db.collection(COLLECTION_SCOOTERS_PATH)
+                    .get()
+                    .get()
+                    .getDocuments()
+                    .stream()
+                    .map(queryDocumentSnapshot -> queryDocumentSnapshot.toObject(Scooter.class))
+                    .collect(Collectors.toSet());
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("FIND ALL SCOOTER - service function.", e);
+            throw new ServiceException(e);
+        }
+
+    }
+
+    public void createScooter(final Scooter scooter) {
+        try {
+            ApiFuture<WriteResult> collectionsApiFuture = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooter.getSerialNumber())
+                    .set(scooter);
+
+            String updateTime = collectionsApiFuture.get().getUpdateTime().toDate().toString();
+
+            LOGGER.info("CREATE SCOOTER - created successfully. scooter = {}. Update_time: {}", scooter, updateTime);
```

Comment:
As you have seen in the article provided in a previous comment, there is a balance between logging enough info in order to be relevant and understand what happened & logging too many details. For example for this log & the below one, how about the following (applies for the other logs as well, but taken this as example since it had logs for success & failure as well):
- For success: `LOGGER.info("Created scooter with id = {} at = {}", scooter, updateTime);`
- For failure: `LOGGER.error("Failed to create scooter with id = {}.", scooter, e);`
Note that those provide the same details, but omit the prefix that you have added, since it is not necessarily relevant, as usually there is a single point where crud operations exist in an app.

Reply:
Agree.